### PR TITLE
Dynamic block reward

### PR DESCRIPTION
**Pull request by paberr**
Implements a dynamic block reward scheme with the following properties:

- **Total Supply:** 21 million NIM
- **Total Supply reached:** after 100 years
- **Initial Reward:** ~5 NIM (4.41074371)
- The reward gradually decreases with the block height, approximately halving every 4 years.
- Starting from block #48696986 (after approximately 92 years), the reward changes from a gradually decreasing curve to a constant reward of 4000 satoshi for the next ~7 years. The previous reward at block #48696985 was 4002 satoshi.

### CODE REVIEW

File: src/main/generic/consensus/Policy.js [hunk 1]
```diff
@@ -36,11 +36,140 @@ class Policy {
...
+     * @param {number} blockHeight
+     * @return {number}
+     */
+    static _blockRewardAt(currentSupply, blockHeight) {
+        const remaining = Policy.TOTAL_SUPPLY - currentSupply;
+        if (blockHeight >= Policy.EMISSION_TAIL_START && remaining >= Policy.EMISSION_TAIL_REWARD) {
+            return Policy.EMISSION_TAIL_REWARD;
+        }
+        const remainder = remaining % Policy.EMISSION_SPEED;
+        return (remaining-remainder) / Policy.EMISSION_SPEED;
```
Comment (sisou) [inline src/main/generic/consensus/Policy.js]: Aren't these two lines the same as
```
return Math.floor(remaining / Policy.EMISSION_SPEED);
```

**Comment by sisou** [comment]
A question from Telegram:
> What's the intention behind the 4000 sat thing? Why not just let it continue until the payout is smaller than half a sat?

File: src/main/generic/consensus/Policy.js (hunk 1 above)
Comment (paberr) [reply src/main/generic/consensus/Policy.js]: In principle, yes, except that the `Math.floor` approach uses a float division. If remaining is large and the division yields a lot of decimal places, floating point inaccuracies might occur.
Our code ensures that the value is divisible by the emission speed and prevents inaccuracies.

**Comment by paberr** [comment]
When using this curve without the 4000 sat tail emission, the normal reward will drop from 1 sat to 0 after 163 years.
At this point, the circulating supply would be 2099999995805697 sat, so 4194303 would be missing.
Emitting these in steps of 1 sat would mean that we reach the total supply of 21mio NIM after 171 years.

The 4000 sat tail emission is used to end up at exactly 21mio NIM after a period of just 100 years (which is already quite far in the future).

**Pull request merged**